### PR TITLE
Backwards compatibility callbacks to authentication APIs

### PR DESCRIPTION
**Pull request by jadahiya-MSFT**

### CODE REVIEW

File: apps/teams-test-app/src/components/AuthenticationAPIs.tsx [hunk 1]
```diff
@@ -17,19 +18,55 @@ const GetAuthToken = (): React.ReactElement =>
...
-      const user = await authentication.getUser();
-      return 'Success: ' + JSON.stringify(user);
+    onClick: {
+      withPromise: async () => {
+        const user = await authentication.getUser();
+        return 'Success: ' + JSON.stringify(user);
+      },
+      withCallback: setResult => {
+        const successCallback = (user: authentication.UserProfile): void => {
+          setResult(JSON.stringify(user));
```
Comment (jadahiya-MSFT) [inline apps/teams-test-app/src/components/AuthenticationAPIs.tsx:58]: I should probably make this consistent with the Promise call by either removing the 'Success' in the withPromise call or adding the 'Success' to this call, I am indifferent to which approach we go with and want to know others feedback

**Comment by erinha** [comment]
<p>Analyzed commit id : 001fb4affc3b4759df3354f6ceac57ff4f1ddfb6</p><hr><details><summary><b><span style="color: green">⯆</span> @microsoft/bundle-analysis-app:</b> -7 Bytes</summary><table><tr><th>Metric Name</th><th>Baseline Size</th><th>Compare Size</th><th>Size Diff</th></tr><tr><td>main.js</td><td>59.38 KB</td><td>59.37 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr><tr><td>Total Size</td><td>59.38 KB</td><td>59.37 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr></table></details><hr><p>Baseline commit: 126f401aae255673ebb77e31804e0ee6bb2cb2f5</p>

File: apps/teams-test-app/src/components/AuthenticationAPIs.tsx [hunk 2]
```diff
@@ -17,19 +18,55 @@ const GetAuthToken = (): React.ReactElement =>
...
-    onClick: async authParams => {
-      const result = await authentication.getAuthToken(authParams);
-      return 'Success: ' + JSON.stringify(result);
+    onClick: {
+      validateInput: () => {
+        return; //This API can have no input
+      },
+      submit: {
+        withPromise: async authParams => {
+          const params = authParams;
```
Comment (MarcinJuraszek) [inline apps/teams-test-app/src/components/AuthenticationAPIs.tsx:27]: is this necessary?

File: apps/teams-test-app/src/components/AuthenticationAPIs.tsx (hunk 1 above)
Comment (MarcinJuraszek) [reply apps/teams-test-app/src/components/AuthenticationAPIs.tsx:58]: Looks like there's no E2E scenario test around this API call, so we should be fine with either. I would probably drop the `success: ` from both.

File: apps/teams-test-app/src/components/AuthenticationAPIs.tsx [hunk 3]
```diff
@@ -63,9 +100,15 @@ const Authenticate = (): React.ReactElement =>
...
-      submit: async authParams => {
-        const token = await authentication.authenticate(authParams);
-        return 'Success: ' + token;
+      submit: {
+        withPromise: async authParams => {
+          const token = await authentication.authenticate(authParams);
+          return 'Success: ' + token;
+        },
+        withCallback: authParams => {
+          authentication.authenticate(authParams);
```
Comment (MarcinJuraszek) [inline apps/teams-test-app/src/components/AuthenticationAPIs.tsx:109]: shouldn't you need to set `authParams.successCallback` and `authParams.failureCallback` before passing them to `authentication.authenticate`?

**Review by MarcinJuraszek — CHANGES_REQUESTED** [review]

File: apps/teams-test-app/src/components/AuthenticationAPIs.tsx (hunk 1 above)
Comment (TrevorJoelHarris) [reply apps/teams-test-app/src/components/AuthenticationAPIs.tsx:58]: Is this also an inconsistency that we should clean up in GetAuthToken?

**Review by TrevorJoelHarris — CHANGES_REQUESTED** [review]
:clock1:

File: apps/teams-test-app/src/components/AuthenticationAPIs.tsx (hunk 3 above)
Comment (jadahiya-MSFT) [reply apps/teams-test-app/src/components/AuthenticationAPIs.tsx:109]: Good catch, you are correct.

**Comment by erinha** [comment]
<p>Analyzed commit id : 93837070ce233ac7915841f763008230d2d2aa8f</p><hr><details><summary><b><span style="color: green">⯆</span> @microsoft/bundle-analysis-app:</b> -7 Bytes</summary><table><tr><th>Metric Name</th><th>Baseline Size</th><th>Compare Size</th><th>Size Diff</th></tr><tr><td>main.js</td><td>59.38 KB</td><td>59.37 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr><tr><td>Total Size</td><td>59.38 KB</td><td>59.37 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr></table></details><hr><p>Baseline commit: 4e3f05ee3569953a280f1b7bba41332ef3db2200</p>

File: apps/teams-test-app/src/components/AuthenticationAPIs.tsx [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import { app, authentication } from '@microsoft/teams-js';
 import React, { ReactElement } from 'react';
 
+import { noHostSdkMsg } from '../App';
```
Comment (Ella-ly) [inline apps/teams-test-app/src/components/AuthenticationAPIs.tsx:4]: I think we never used `noHostSdkMsg` in this file.

**Review by TrevorJoelHarris — APPROVED** [review]
:shipit:

**Review by MarcinJuraszek — APPROVED** [review]

**Pull request merged**